pt_miniscreen/core/app.py

```diff
@@ -0,0 +1,26 @@
+import logging
+
+from PIL import Image
+
+logger = logging.getLogger(__name__)
+
+
+class App:
+    def __init__(self, miniscreen=None, Root=None):
+        assert miniscreen is not None
+        assert Root is not None
+        self.miniscreen = miniscreen
+        self.Root = Root
+
+    def start(self):
+        self.root = self.Root(on_rerender=self.display)
+        self.display()
```

Comment:
should the components call on_rerender themselves when they're first created?

Reply:
No I don't think so. If you make a component call rerender on creation it does not allow for you to properly set up all the children before render is called. It's also not guaranteed that a child will be used by the parent in it's render method, it might be hidden by default for example. If we did get around those aspects it would still result in unnecessary reconciles.